### PR TITLE
Fix failing tests.

### PR DESCRIPTION
**Pull request by glenpike**
Move 2 email regex tests from invalid to valid #489
Fix problems with adding list items - spy on 'initialize' instead
of Class constructor, possibly fallout from #451
List.NestedModel getStringValue test was failing on alternate runs,
the previous test messed with the prototype, which seemed to
cause the fail - restore the proto after test.

### CODE REVIEW

**Comment by glenpike** [comment]
@exussum12 can you review and merge if all's good please?

File: test/editors/extra/list.js [hunk 1]
```diff
@@ -1338,6 +1343,8 @@ test('getStringValue() - uses model.toString() if available', function() {
 
     this.editor.setValue({ id: 1, name: 'foo' });
 
+    console.log('editor ', this.editor);
```
Comment (exussum12) [inline test/editors/extra/list.js]: Can the console log be removed ?

**Comment by exussum12** [comment]
Changes look good!

**Comment by glenpike** [comment]
@exussum12 removed the console.log

**Comment by exussum12** [comment]
Merged. Good work

**Pull request merged**